File: Assets/Scripts/MainPage.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using LuaInterface;

public class MainPage : MonoBehaviour {

    private LuaState lua;
    private LuaFunction func = null;
    public GameObject MainPagePrefab;

    void Awake()
    {
        lua = new LuaState();    //lua解析器
        lua.Start();     //启动lua
        string luaPath = Application.dataPath + "\\Scripts/";
        lua.AddSearchPath(luaPath);
        lua.Require("MainPage");   //Require读取lua文件只执行一次
        func = lua.GetFunction("MainPage.Awake");
        if (func != null)
        {
            CallLuaFunction();
        }

        lua.CheckTop();
        lua.Dispose();
    }
	// Use this for initialization
	void Start () {
		
	}
	
	// Update is called once per frame
	void Update () {
		
	}

    void CallLuaFunction()
    {
        func.BeginPCall();
        func.Push(MainPagePrefab.transform);
        func.PCall();
        func.EndPCall();
    }
}

File: Assets/Scripts/MainPage.lua
MainPage = {}
local this  = MainPage
local gameObject
local transform

print("load MainPage.lua")

function MainPage.Awake(obj)
	gameObject = obj   --得到gameObject对象
	transform = obj.transform

	this.Init()
	this.PrintInfo()
end

function MainPage.Init()
	this.btn_StartGame = transform:Find("btn_StartGame").gameObject
	this.btn_ExitGame = transform:Find("btn_ExitGame").gameObject
end

function MainPage.PrintInfo()
	print("btn_StartGame is : "..this.btn_StartGame)
	print("btn_ExitGame is : "..this.btn_ExitGame)
end